lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -181,23 +182,31 @@ def load_config(self, config, return_error=False, opts=None):
         if opts is None:
             opts = {}
 
-        rc, out, err = self.exec_command('configure')
-        if rc != 0:
-            self._module.fail_json(msg='unable to enter configuration mode', output=to_text(err))
+        connection = self._get_connection()
 
         msgs = []
-        for cmd in config:
-            rc, out, err = self.exec_command(cmd)
-            if opts.get('ignore_timeout') and rc == 1:
-                msgs.append(err)
-                return msgs
-            elif rc != 0:
-                self._module.fail_json(msg=to_text(err))
-            elif out:
-                msgs.append(out)
-
-        self.exec_command('end')
-        return msgs
+        try:
+            responses = connection.edit_config(config)
+            out = json.loads(responses)[1:-1]
+            msg = out
+        except ConnectionError as e:
+            code = getattr(e, 'code', 1)
+            message = getattr(e, 'err', e)
+            err = to_text(message, errors='surrogate_then_replace')
+            if opts.get('ignore_timeout') and code == 1:
+                return msgs.append(err)
+            elif code != 0:
+                self._module.fail_json(msg=err)
+
+        return msgs.append(msg)
+
+    def get_capabilities(self):
```

Comment:
`capabilities` can be cached instead of fetching it from remote host each time.
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/vyos/vyos.py#L80